app/screens/login/form.tsx

```diff
@@ -238,13 +306,21 @@ const LoginForm = ({config, extra, keyboardAwareRef, serverDisplayName, launchEr
         setEmmUsernameIfAvailable();
     }, [managedConfig?.username]);
 
-    useEffect(() => {
-        if (loginId && password) {
-            setButtonDisabled(false);
+    const onIdInputSubmitting = useCallback(() => {
+        if (!magicLinkEnabled || (userLoginType !== 'guest_magic_link')) {
+            focusPassword();
             return;
         }
-        setButtonDisabled(true);
-    }, [loginId, password]);
+
+        onLogin();
+    }, [focusPassword, onLogin, magicLinkEnabled, userLoginType]);
+
+    const buttonDisabled = getButtonDisabled(loginId, password, userLoginType, magicLinkEnabled);
+    const showPasswordInput = !magicLinkEnabled || (userLoginType !== 'guest_magic_link' && userLoginType !== undefined && userLoginType !== 'deactivated');
+    let userInputError = error;
+    if (showPasswordInput) {
+        userInputError = error ? ' ' : '';
```

Comment:
should there be an error string? if not, at least add a comment as this is a bit weird

I see this was a refactor to make things clearer, but still